marshal/tests/serialization/set.go

```diff
@@ -0,0 +1,104 @@
+package serialization
+
+import (
+	"bytes"
+	"fmt"
+	"runtime/debug"
+	"testing"
+
+	"github.com/gocql/gocql/marshal/tests/funcs"
+	"github.com/gocql/gocql/marshal/tests/utils"
+)
+
+type Sets []*Set
+
+// Set is a tool for generating test cases of marshal and unmarshall funcs.
+// For cases when the function should no error,
+// marshaled data from Set.Values should be equal with Set.Data,
+// unmarshalled value from Set.Data should be equal with Set.Values.
+type Set struct {
+	Data   []byte
+	Values []interface{}
+
+	IssueMarshal   string
+	IssueUnmarshal string
+}
+
+func (s Set) Run(name string, t *testing.T, marshal func(interface{}) ([]byte, error), unmarshal func([]byte, interface{}) error) {
+	if name == "" {
+		name = utils.ValueToName(s.Values[0])
+	}
+
+	t.Run(name, func(t *testing.T) {
+		for i := range s.Values {
+			val := s.Values[i]
+
+			t.Run(fmt.Sprintf("%T", val), func(t *testing.T) {
+				if marshal != nil {
+					s.runMarshal(t, marshal, val)
+				}
+
+				if unmarshal != nil {
+					s.runUnmarshal(t, unmarshal, val)
+				}
+			})
+		}
+	})
+}
+
+func (s Set) runMarshal(t *testing.T, f func(interface{}) ([]byte, error), val interface{}) {
+	t.Run("marshal", func(tt *testing.T) {
```

Comment:
No need to have `tt` here, you can safely override `t`:
```suggestion
	t.Run("marshal", func(t *testing.T) {
```

Reply:
Tested... t.Skip skips original t.

Reply:
what is your go version ?